### PR TITLE
nav2 profile: part 1: pgp

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] search as popup
- [x] logged out override header border
- [x] no tracker cards
- [x] pgp
- [x] revoke proof

### CODE REVIEW

File: shared/actions/profile/index.js [hunk 1]
```diff
@@ -79,15 +80,32 @@ const onClickAvatar = (_, action) => {
...
+        flags.useNewRouter ? null : RouteTreeGen.createNavigateTo({parentPath: [peopleTab], path: []})
+      )
+      .catch(e => {
+        logger.info('error in dropping pgp key', e)
+        return ProfileGen.createRevokeFinishError({error: `Error in dropping Pgp Key: ${e}`})
       })
-    })
+  } else {
+    return RPCTypes.revokeRevokeSigsRpcPromise({sigIDQueries: [action.payload.proofId]}, Constants.waitingKey)
+      .then(() => ProfileGen.createFinishRevoking())
```
Comment (buoyad) [inline shared/actions/profile/index.js]: `finishRevoking` causes a `navigateUp`, but you also clear modals [here](https://github.com/keybase/client/pull/16506/files#diff-dd23192500e15800dc578aa31c3a3c80R30) so you end up at the root of the people tab when revoking a non-pgp.

File: shared/profile/routes.js [hunk 2]
```diff
@@ -96,22 +96,22 @@ const profileRoute = () => {
...
+  profileProveEnterUsername: {getScreen: () => require('./prove-enter-username/container').default},
+  profileProveWebsiteChoice: {getScreen: () => require('./prove-website-choice/container').default},
+  profileShowcaseTeamOffer: {getScreen: () => require('./showcase-team-offer/container').default},
 }
 
 export const newModalRoutes = {
-  search: {getScreen: () => require('./search/container').default},
+  profileEdit: {getScreen: () => require('./edit-profile2/container').default},
+  profileRevoke: {getScreen: () => require('./revoke/container').default},
+  profileSearch: {getScreen: () => require('./search/container').default},
```
Comment (buoyad) [inline shared/profile/routes.js]: is this supposed to work on mobile?

File: shared/router-v2/router.desktop.js [hunk 3]
```diff
@@ -50,7 +50,7 @@ class AppView extends React.PureComponent<any> {
           fullHeight={true}
           style={selectedTab ? styles.contentArea : styles.contentAreaLogin}
         >
-          <Header options={descriptor.options} onPop={childNav.pop} allowBack={index !== 0} />
+          <Header options={descriptor.options} onPop={() => childNav.pop()} allowBack={index !== 0} />
```
Comment (buoyad) [inline shared/router-v2/router.desktop.js]: what was the problem here?

File: shared/router-v2/router.desktop.js [hunk 4]
```diff
@@ -278,6 +284,7 @@ const styles = Styles.styleSheetCreate({
   }),
   modalContainer: {
     ...Styles.globalStyles.fillAbsolute,
+    zIndex: 99999,
```
Comment (buoyad) [inline shared/router-v2/router.desktop.js]: can we avoid this? this is problematic for modals that are supposed to have floating stuff on top of them.

File: shared/profile/routes.js [hunk 5]
```diff
@@ -96,22 +96,22 @@ const profileRoute = () => {
...
-  editProfile2: {getScreen: () => require('./edit-profile2/container').default},
-  nonUserProfile: {getScreen: () => require('./non-user-profile/container').default},
-  postProof: {getScreen: () => require('./post-proof/container').default},
   profile: {getScreen: () => require('./user/container').default, upgraded: true},
-  proveEnterUsername: {getScreen: () => require('./prove-enter-username/container').default},
-  proveWebsiteChoice: {getScreen: () => require('./prove-website-choice/container').default},
-  revoke: {getScreen: () => require('./revoke/container').default},
-  showcaseTeamOffer: {getScreen: () => require('./showcase-team-offer/container').default},
-  ...require('./pgp/routes').newRoutes,
+  profileAddToTeam: {getScreen: () => require('./add-to-team/container').default},
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/profile/routes.js]: yup, there's going to be part 2/3/4 etc

File: shared/profile/pgp/info.desktop.js [hunk 6]
```diff
@@ -5,33 +5,15 @@ import * as Kb from '../../common-adapters'
...
-const Info = (props: Props) => {
-  const nextDisabled = !props.email1 || !props.fullName || !!props.errorText
-  return (
-    <Kb.StandardScreen style={styleContainer} onCancel={props.onCancel}>
-      {/* TODO(MM) when we get the pgp icon, put it in here */}
-      <Kb.PlatformIcon platform="pgp" overlay="icon-proof-unfinished" style={styleIcon} />
-      <Kb.Text style={styleHeader} type="BodySemibold">
-        {' '}
+const Info = props => (
+  <Modal>
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/profile/pgp/info.desktop.js]: yeah i have a fix for this in part 2. lets merge that on top before this gets into master

File: shared/actions/profile/index.js (hunk 1 above)
Comment (chrisnojima) [reply shared/actions/profile/index.js]: this is fixed in part 2

File: shared/profile/routes.js (hunk 2 above)
Comment (chrisnojima) [reply shared/profile/routes.js]: yes

File: shared/router-v2/router.desktop.js (hunk 3 above)
Comment (chrisnojima) [reply shared/router-v2/router.desktop.js]: onPop passes an event, childNav.pop plumbs through params so its possible you can do childNav.pop(3) which goes back 3 levels or whatever

File: shared/router-v2/router.desktop.js (hunk 4 above)
Comment (chrisnojima) [reply shared/router-v2/router.desktop.js]: we should discuss a solution. on profile there was some absolutely positioned stuff due to the section list which goes above the modal

File: shared/people/index.desktop.js [hunk 7]
```diff
@@ -2,49 +2,63 @@
...
+          People
+        </Kb.Text>
+        <ProfileSearch />
+      </Kb.Box2>
+    )
+  : (props: Props) => (
+      <Kb.HeaderHocHeader
+        headerStyle={styles.header}
+        rightActions={
+          flags.useNewRouter
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/people/index.desktop.js]: fixed in part 2

File: shared/profile/pgp/info.desktop.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (buoyad) [reply shared/profile/pgp/info.desktop.js]: ok I'll wait till that PR is ready to review both together

**Pull request merged**